src/Parser/Priest/Discipline/Modules/Spells/Contrition.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { formatPercentage, formatNumber } from 'common/format';
+import Analyzer from 'Parser/Core/Analyzer';
+
+import Penance from '../Spells/Penance';
+import { OffensivePenanceBoltEstimation } from '../../SpellCalculations';
+
+class Contrition extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    statTracker: StatTracker,
+    penance: Penance, // we need this to add `penanceBoltNumber` to the damage and heal events
+  };
+
+  healing = 0;
+  damagePenalty = 0;
+  player = null;
+  penanceBoltEstimation;
+
+  on_initialized() {
+    this.active = this.owner.modules.combatants.selected.hasTalent(SPELLS.CONTRITION_TALENT.id);
+    this.penanceBoltEstimation = OffensivePenanceBoltEstimation(this.statTracker);
+  }
+
+  /**
+   * Theory behind this is as follows
+   * By casting a contrition penance, you lose an offensive penance worth of healing
+   * To hone in on the true value of contrition, we estimate the healing of an offensive
+   * penance hit, and subtract that from the contrition amount.
+   *
+   * We keep the penance heal as that is a true gain from choosing a contrition
+   * penance over a regular offensive one.
+   */
+  on_byPlayer_heal(event) {
+    if (event.ability.guid !== SPELLS.CONTRITION_HEAL.id && event.ability.guid !== SPELLS.PENANCE_HEAL.id) {
+      return;
+    }
+
+    // Add the healing to our count
+    this.healing += event.amount;
+
+    // Get an estimated amount of damage and healing for a bolt
+    const { damage, healing } = this.penanceBoltEstimation();
+
+    // Calculate the difference between contrition and an offensive penance
+    if (event.ability.guid === SPELLS.CONTRITION_HEAL.id) {
+      const overhealing = healing - event.overheal;
```

Comment:
Shouldn't overhealing only be considered if `healing > event.amount`, and in that case just set healing = event.amount?
for instance, if contrition heals for 50 and overheals for 50 and atonement would have healed for 75 (50 + 25 overheal), then it should just subtract the full value of heal for a net 0 gain from speccing contrition right?

`this.healing -= estimatedDifference;` unless I misunderstood it, you should probably rename estimatedDifference to atonmentHeal or something like that. I assume that's what it's meant to be atleast since you add the full contrition heal in l48, it would make sense to subtract the full atonement heal instead of the difference.